src/esp/sensor/CameraSensor.cpp

```diff
@@ -115,20 +124,30 @@ bool CameraSensor::drawObservation(sim::Simulator& sim) {
   renderTarget().renderEnter();
 
   gfx::RenderCamera::Flags flags;
-  if (sim.isFrustumCullingEnabled())
+  if (sim.isFrustumCullingEnabled()) {
     flags |= gfx::RenderCamera::Flag::FrustumCulling;
+  }
 
-  gfx::Renderer::ptr renderer = sim.getRenderer();
   if (cameraSensorSpec_->sensorType == SensorType::Semantic) {
     // TODO: check sim has semantic scene graph
-    renderer->draw(*this, sim.getActiveSemanticSceneGraph(), flags);
-    if (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph()) {
+    bool twoSceneGraphs =
+        (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph());
+
+    if (twoSceneGraphs) {
+      moveSemanticSensorToSemanticSceneGraph(sim);
```

Comment:
This seems a bit risky as it's easy to forget to call the `moveSemanticSensorBackToRegularSceneGraph` function.  Worth considering RAII here.

Reply:
Can you elaborate a bit how you are going to do RAII here?

if people forgets to call this function `moveSemanticSensorBackToRegularSceneGraph`, she will know immediately as in the following (L145), she needs it to draw the "objects". If the above function is not called, the program will hit an assertion in Magnum (camera is not in the same SG) and quit.

Reply:
Just a helper class that calls `moveSemanticSensorToSemanticSceneGraph` in its constructor and `moveSemanticSensorBackToRegularSceneGraph` in its destructor.